### PR TITLE
Improve ServerCertificatesPerAccount limit error message

### PR DESCRIPTION
**Pull request by niklasvincent**
One of the problem that can occur during the certificate upload process is hitting the `ServerCertificatesPerAccount` limit, which means you have to request an increase to the number of SSL certificates you can store in your AWS account.

The error could be improved a bit:

```
com.amazonaws.services.identitymanagement.model.LimitExceededException: Cannot exceed quota for ServerCertificatesPerAccount: 20 (Service: AmazonIdentityManagement; Status Code: 409; Error Code: LimitExceeded; Request ID: XYZ)
com.amazonaws.services.identitymanagement.model.LimitExceededException: Cannot exceed quota for ServerCertificatesPerAccount: 20 (Service: AmazonIdentityManagement; Status Code: 409; Error Code: LimitExceeded; Request ID: XYZ)
```

This change makes the error message a bit more user friendly and points the user in the right direction as to how to resolve it. It also opens up for catching and handling more exceptions that could happen during the certificate upload process.

### CODE REVIEW

**Comment by adamnfish** [comment]
This is a very useful change! I've not tested it but looks good to me :+1:

File: src/main/scala/com/gu/certificate/Magic.scala [hunk 1]
```diff
@@ -90,15 +90,20 @@ object Magic extends BouncyCastle with FileHelpers {
...
+    val certificateUploadRequest = new UploadServerCertificateRequest()
+      .withServerCertificateName(s"$safeDomain-exp$expDate")
+      .withPrivateKey(decryptedPem)
+      .withCertificateBody(certificatePem)
+      .withCertificateChain(chainPem)
+
+    Try {
+      val certificateUploadResult = iamClient.uploadServerCertificate(certificateUploadRequest)
+      val certificateArn = certificateUploadResult.getServerCertificateMetadata.getArn
+      System.err.println(s"successfully installed certificate in IAM as ${certificateArn}")
```
Comment (mchv) [inline src/main/scala/com/gu/certificate/Magic.scala:102]: very small and was already there, but why `System.err` and not `System.out` when the installation is successful?

Comment (niklasvincent) [inline src/main/scala/com/gu/certificate/Magic.scala:102]: Good question. I just kept with the convention already in the code. I wonder if it is to do with how SBT treats the output, but @sihil would know.

Comment (sihil) [inline src/main/scala/com/gu/certificate/Magic.scala:102]: For CLI apps I generally use the convention that if the output is potentially processable by another app then it should go to stdout. For text that would get in the way of parsing output then it should go to stderr. Take `ps` as an example - the column headers go to stderr and the data to stdout. Prism is similar, the host count and header go to stderr and the lines of hosts to stdout.

In this case there isn't really anything of use to send to stdout. Although the ARN of the cert is arguably one piece of information you might want to grab in a script.

Comment (niklasvincent) [inline src/main/scala/com/gu/certificate/Magic.scala:102]: Makes perfect sense. Thanks!

**Comment by niklasvincent** [comment]
Anyone with merge permissions (e.g. not me), feel free to merge.

**Comment by sihil** [comment]
Not sure what the _"release"_ process is so will leave this to @adamnfish

**Comment by adamnfish** [comment]
OK :-)

**Comment by adamnfish** [comment]
https://github.com/guardian/certificate-magic/releases

The README points at `latest` so this is now released.

**Pull request merged**